### PR TITLE
4570: JS code style in webtrekk

### PR DESCRIPTION
**Pull request by cableman**
#### Link to issue

https://platform.dandigbib.org/issues/4570

#### Description

Minor code style changes made during JS debug.

#### Screenshot of the result

No screenshot.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

There is no changes in the code, only code style.

### CODE REVIEW

File: modules/ding_webtrekk/ding_webtrekk.js [hunk 1]
```diff
@@ -154,11 +159,11 @@
         });
 
         $('.slick-arrow', this).once('js-ding-webtrekk').click(function() {
-          var linkId = 'Karousel, click på forrige knappen';
+          var linkId = Drupal.t('Carousel, clicked on previous link');
           var wtkId = DING_WEBTREKK_PARAMETER_CAROUSEL_PREV;
 
           if ($(this).hasClass('slick-next')) {
-            linkId = 'Karousel, click på næste knappen';
```
Comment (kasperg) [inline modules/ding_webtrekk/ding_webtrekk.js]: As I understand it these are strings used by administrators in Webtrekk - not end users on the site. Is there a reason why they should pass through the translation engine?

If someone changes the translation without knowing where it is sed I fear will cause unwanted changes in Webtrekk reporting.

File: modules/ding_webtrekk/ding_webtrekk.js [hunk 2]
```diff
@@ -21,13 +20,13 @@
   var DING_WEBTREKK_PARAMETER_CAROUSEL_NEXT = 59;
   var DING_WEBTREKK_PARAMETER_CAROUSEL_PREV = 60;
 
-  var appendQueryParameter = function(url, key, value) {
```
Comment (kasperg) [inline modules/ding_webtrekk/ding_webtrekk.js]: Why do we need to name our functions when we also assign them to variables?

File: modules/ding_webtrekk/ding_webtrekk.js (hunk 1 above)
Comment (cableman) [reply modules/ding_webtrekk/ding_webtrekk.js]: K, it's just wrong to use danish inside JS code.

File: modules/ding_webtrekk/ding_webtrekk.js (hunk 2 above)
Comment (cableman) [reply modules/ding_webtrekk/ding_webtrekk.js]: So when it don't work the console will write which function the error happened in a not just anonymous function.

It makes debugging easier when error message gets copied into a ticket.

File: modules/ding_webtrekk/ding_webtrekk.js (hunk 1 above)
Comment (cableman) [reply modules/ding_webtrekk/ding_webtrekk.js]: But, reverted back.

File: modules/ding_webtrekk/ding_webtrekk.js (hunk 2 above)
Comment (kasperg) [reply modules/ding_webtrekk/ding_webtrekk.js]: Thanks for elaborating. I think that is a totally valid argument. In that case: Have you considered whether you need to assign `function appendQueryParameter` to a variable at all?

I think it should be possible to declare the named functions without the assignment and then call them as you would normally.

Comment (cableman) [reply modules/ding_webtrekk/ding_webtrekk.js]: The issue here is that they are "exported" into global "Drupal" space in line 43 (https://github.com/ding2/ding2/blob/74e40bb8b9c0d36c897d1359598dacb4d20e243d/modules/ding_webtrekk/ding_webtrekk.js#L43) if they where not I would have removed the assignment.

Comment (holt83) [reply modules/ding_webtrekk/ding_webtrekk.js]: It's perfectly fine and normal to have a named function expression: https://kangax.github.io/nfe/#named-expr

Comment (kasperg) [reply modules/ding_webtrekk/ding_webtrekk.js]: @holt83 I am not arguing against it.

As I see it @cableman is trying to [improve debugging for named functions](https://kangax.github.io/nfe/#names-in-debuggers) while my suggestion was then to simplify the existing code.

I see that the simplification is not possible so lets just roll with this.

Comment (holt83) [reply modules/ding_webtrekk/ding_webtrekk.js]: @kasperg I think that is the way to go!

Maybe we could also prefix the function names with dingWebtrekk to make it more clear that this is ding_webtrekk functions when they appear in debugging. But it's a minor one.

**Pull request merged**